File: SmartRoutes/Scripts/Communication/Payloads/ServiceTypePayload.js

// This is the payload object that 
SmartRoutes.Communication.ServiceTypePayload = function(serviceTypeViewModel) {

    return {
        // The name for the service type.
        Name: serviceTypeViewModel.name(),
        // True if the service type is checked, false otherwise.
        Checked: serviceTypeViewModel.checked()
    }
};

File: SmartRoutes/Models/Payloads/ServiceTypePayload.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SmartRoutes.Models.Payloads
{
    /// <summary>
    /// Payload object representing a service type to use in the search.
    /// </summary>
    public class ServiceTypePayload
    {
        /// <summary>
        /// Constructor.
        /// </summary>
        public ServiceTypePayload()
        {

        }

        /// <summary>
        /// Name of the service type.
        /// </summary>
        public string Name
        {
            get;
            set;
        }

        /// <summary>
        /// Flag indicating if the service type is checked.
        /// </summary>
        public bool Checked
        {
            get;
            set;
        }
    }
}